src/main/java/javacake/storage/Storage.java

```diff
@@ -0,0 +1,181 @@
+package javacake.storage;
+
+import javacake.Duke;
+import javacake.exceptions.DukeException;
+import javacake.tasks.Task;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileReader;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.StringTokenizer;
+import java.util.logging.Level;
+
+public class Storage {
+    private int stringBuffer = 7;
+    private static ArrayList<Task> internalTaskData = new ArrayList<>();
+
+    private String filepath;
+    private TaskType dataType;
+    public TaskList tasks;
+
+    public enum TaskType {
+        TODO, DEADLINE, TODO_DAILY, TODO_WEEKLY, TODO_MONTHLY
+    }
+
+    /**
+     * Constructor for storage.
+     */
+    public Storage() throws DukeException {
+        this.tasks = new TaskList();
+        //Initialise new deadline file
+        try {
+            File tasksFile = new File("data/tasks/deadline.txt");
+            filepath = tasksFile.getPath();
+            Duke.logger.log(Level.INFO,"Filepath: " + filepath);
+
+            if (!tasksFile.getParentFile().getParentFile().exists()) {
+                tasksFile.getParentFile().getParentFile().mkdir();
+                tasksFile.getParentFile().mkdir();
+                tasksFile.createNewFile();
+                System.out.println("A" + tasksFile.getParentFile().getParentFile().getPath());
+            } else if (!tasksFile.getParentFile().exists()) {
+                tasksFile.getParentFile().mkdir();
+                tasksFile.createNewFile();
+                System.out.println("B" + tasksFile.getParentFile().getPath());
+            } else if (!tasksFile.exists()) {
+                tasksFile.createNewFile();
+                System.out.println("C" + tasksFile.getPath());

```

Comment:
This would be a good place to use a logger, as opposed to system io, since your team is already using one.